### DISCUSSION TITLE
A new potential developer, wishing to implement video preview thumbnails and fuzzy search.

### DISCUSSION THREAD

**Discussion by SnowblindFatal** [jellyfin/jellyfin#9737] [General]
Hello! I work as a professional software developer but haven't really participated in open source projects thus far. I'd like to see some features in Jellyfin, and I figured I might as well give it a shot to implement them myself since I'm sure you folks are busy enough as it is.

Feature request for video preview: https://features.jellyfin.org/posts/139/show-video-preview-on-thumbnail-hover
I was thinking about cycling through thumbnail screenshots of the video file instead of cycling short video snippets to keep it simple. Maybe the chapter images could be used for this purpose, but I'm not exactly sure about their working principles, so I'm not sure how usable the images would be. Alternatively, I could extract my own set of thumbnail screenshots from the video for the preview shots.

The second thing I would like to implement is fuzzy search. https://features.jellyfin.org/posts/1141/allow-typos-misspellings-in-search seems like a pretty good match. Also, while at it, I would like to include tags in the search matching and not just in filtering options. Additionally, I'd like to improve the search results presentation so that the results are displayed in a tiled view or a list instead of a single row of a few thumbnails.

Given that I don't know the architecture of Jellyfin much at all, some pointers on where to start modding would be most welcome. Likewise, it would be helpful to know if these features should be integrated into the main project or be implemented as plugins.

**Comment by nyanmisaka** [comment]
1) https://github.com/jellyfin/jellyfin/pull/9554

**Reply by SnowblindFatal** [reply]
Thanks for the response. Glad to see this is already being tackled. Looks like the fuzzy search feature as well as the search results improvement still needs implementation.

**Comment by justin13888** [comment]
Wondering if @SnowblindFatal or others have started working on Fuzzysearch as that is a feature I would also like and thinking of implementing myself.
